### PR TITLE
doc: matter: add device type page

### PR DESCRIPTION
**Pull request by greg-fer**
Added device type page.
KRKNWK-15343.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_matter_overview_device_types.rst [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+|                   | of measuring and reporting the occupancy state in a designated area.         |                     |                                       |
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Temperature       | A Temperature Sensor device reports measurements of temperature.             | Certifiable         | :ref:`matter_weather_station_app`     |
+| Sensor            |                                                                              |                     |                                       |
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Pressure Sensor   | A Pressure Sensor device measures and periodically reports the pressure      | Provisional         | :ref:`matter_weather_station_app`     |
+|                   | of a fluid.                                                                  |                     |                                       |
```
Comment (kkasperczyk-no) [inline doc/nrf/ug_matter_overview_device_types.rst:155]: I'm not sure about this pressure of a fluid. I can see that this is what the device library document says, but in our case sensor measures atmospheric pressure.

File: doc/nrf/ug_matter_overview_device_types.rst [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+A device type can require other device types for its composition, making it a *composed device type*.
+
+Device type overview
+********************
+
+The following tables list the application device types that are supported in the Matter specification.
+
+* The description for each device type is taken from the Device Library specification.
+* The state of the device specifies whether a device type can be subject to certification.
+  The `Provisional` state indicates that the device type is not yet ready for implementation.
```
Comment (kkasperczyk-no) [inline doc/nrf/ug_matter_overview_device_types.rst:33]: I would try to rephrase it a little bit to better describe the situation for a user. Provisional state doesn't describe the implementation state, but rather certification state. It means that implementation is most likely ready, but this device type didn't get enough coverage (to small amount of people tested it) to claim it is certified. Would something like this suite you:
```suggestion
  The `Provisional` state indicates that the device type implementation was not yet fully tested and certified, however implementation may be ready and can be used by a user on his own risk.
```

File: doc/nrf/ug_matter_overview_device_types.rst [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+.. _ug_matter_device_types_hvac:
+
+HVAC device types
+=================
+
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Device type       | Description (from Device Library specification)                              | State of the device | Dedicated sample in the |NCS|         |
++===================+==============================================================================+=====================+=======================================+
+| Heating/Cooling   | A Heating/Cooling Unit is a device capable of heating or cooling a space     | Certifiable         |                                       |
```
Comment (kkasperczyk-no) [inline doc/nrf/ug_matter_overview_device_types.rst:197]: @doublemis1 could you confirm that? I think I heard about adding support to this device type in Matter 1.1, so I'm not sure if that was certified. The same applies to Fan controller.

File: doc/nrf/ug_matter_overview_device_types.rst [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+|                   | the color of the light being emitted. In addition, a Control Bridge device   |                     |                                       |
+|                   | is capable of being used for setting scenes.                                 |                     |                                       |
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Pump Controller   | A Pump Controller device is capable of configuring and controlling           | Certifiable         |                                       |
+|                   | a Pump device.                                                               |                     |                                       |
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Generic Switch    | General-purpose switch that can have more than two positions.                | Certifiable         |                                       |
```
Comment (doublemis1) [inline doc/nrf/ug_matter_overview_device_types.rst:128]: @kkasperczyk-no Do we support the generic switch only on upstream only?

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 3 above)
Comment (doublemis1) [reply doc/nrf/ug_matter_overview_device_types.rst:197]: You are right:
List of Provisional Items
The following is a list of provisional items.
• Support for Heating/Cooling Unit is provisional.
• Support for Fan device type is provisional.

Need to verify if the pressure sensor is set to provisional

File: doc/nrf/ug_matter_overview_device_types.rst [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+| Occupancy Sensor  | An Occupancy Sensor is a measurement and sensing device that is capable      | Certifiable         |                                       |
+|                   | of measuring and reporting the occupancy state in a designated area.         |                     |                                       |
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Temperature       | A Temperature Sensor device reports measurements of temperature.             | Certifiable         | :ref:`matter_weather_station_app`     |
+| Sensor            |                                                                              |                     |                                       |
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Pressure Sensor   | A Pressure Sensor device measures and periodically reports the pressure      | Provisional         | :ref:`matter_weather_station_app`     |
```
Comment (doublemis1) [inline doc/nrf/ug_matter_overview_device_types.rst:154]: Based on current spec the Pressure Sensor is not provisional

File: doc/nrf/ug_matter_overview_device_types.rst [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+.. _ug_matter_device_types_lighting:
+
+Lighting device types
+=====================
+
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Device type       | Description (from Device Library specification)                              | State of the device | Dedicated sample in the |NCS|         |
++===================+==============================================================================+=====================+=======================================+
+| On/Off Light      | The On/Off Light is a lighting device that is capable of being switched on   | Certifiable         | :ref:`matter_light_bulb_sample`       |
```
Comment (doublemis1) [inline doc/nrf/ug_matter_overview_device_types.rst:46]: The matter_light_bulb_sample is only a Dimmable Light
AFAIK, the device shall support one application device type per endpoint, our sample supports all necessary clusters to be validated as Dimmable Light

File: doc/nrf/ug_matter_overview_device_types.rst [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+.. _ug_matter_device_types_switches_controls:
+
+Switches and Controls device types
+==================================
+
++-------------------+------------------------------------------------------------------------------+---------------------+---------------------------------------+
+| Device type       | Description (from Device Library specification)                              | State of the device | Dedicated sample in the |NCS|         |
++===================+==============================================================================+=====================+=======================================+
+| On/Off Light      | An On/Off Light Switch is a controller device that,                          | Certifiable         | :ref:`matter_light_switch_sample`     |
```
Comment (doublemis1) [inline doc/nrf/ug_matter_overview_device_types.rst:103]: The same story as for the Light bulb.
The Light switch supports all necessary clusters to be validated as Dimmer Switch.

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 1 above)
Comment (doublemis1) [reply doc/nrf/ug_matter_overview_device_types.rst:155]: The latest Device Library Matter spec defines pressure sensors as follows:
7.5. Pressure Sensor
A Pressure Sensor device measures and reports the pressure of a fluid.

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_device_types.rst:33]: Rephrased slightly.

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_device_types.rst:155]: No changes made. Should we remove the mention of the weather app in this line if we're not matching the device type description, @kkasperczyk-no , @doublemis1 ?

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 3 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_device_types.rst:197]: Changed both to Provisional.

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 5 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_device_types.rst:154]: Changed.

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 6 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_device_types.rst:46]: Removed as sample for On/Off Light.

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 7 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_device_types.rst:103]: Removed as example for On/Off Light Switch.

**Comment by greg-fer** [comment]
@kkasperczyk-no , @doublemis1 , edits applied. Please check.

File: doc/nrf/ug_matter_overview_device_types.rst (hunk 1 above)
Comment (doublemis1) [reply doc/nrf/ug_matter_overview_device_types.rst:155]: I will keep Weather Station as it is since the fluid is a Liquid or Gas

**Review by doublemis1 — APPROVED** [review]
LGTM!

**Review by kkasperczyk-no — APPROVED** [review]

**Comment by greg-fer** [comment]
Changelog entry handled in https://github.com/nrfconnect/sdk-nrf/pull/8969

**Pull request merged**